ls1intum/Artemis
# [Enhancement] Security & Auto-approve answers by instructors

### Checklist

### Motivation and Context


An answer given by an instructor is not highlighted except if it is approved manually.
There are insufficient checks for security with the questions and answers.

### Description

Answers given by an instructor are now approved automatically.
Include course in api calls and check the user's access right in all endpoints.


### Steps for Testing


1. Log in to Artemis
2. Navigate to an exercise or lecture with a question
3. Write answers with different access rights (e.g student, instructor)
4. Check that only answers given by instructors are approved automatically.

### Test Coverage

* StudentQuestionAnswerResource.java: 92%
